apps/ios/SimpleXChat/ChatTypes.swift

```diff
@@ -2243,32 +2243,62 @@ public struct CIFile: Decodable {
             case .sndTransfer: return true
             case .sndComplete: return true
             case .sndCancelled: return true
+            case .sndError: return true
             case .rcvInvitation: return false
             case .rcvAccepted: return false
             case .rcvTransfer: return false
             case .rcvCancelled: return false
             case .rcvComplete: return true
+            case .rcvError: return false
             }
         }
     }
 
-    public var cancellable: Bool {
+    public var cancellableText: (String, String, String, String)? {
         get {
             switch self.fileStatus {
-            case .sndStored: return self.fileProtocol != .xftp // TODO true - enable when XFTP send supports cancel
-            case .sndTransfer: return self.fileProtocol != .xftp // TODO true
-            case .sndComplete: return false
-            case .sndCancelled: return false
-            case .rcvInvitation: return false
-            case .rcvAccepted: return true
-            case .rcvTransfer: return true
-            case .rcvCancelled: return false
-            case .rcvComplete: return false
+            case .sndStored: return sndCancelTexts
+            case .sndTransfer: return sndCancelTexts
+            case .sndComplete:
+                if self.fileProtocol == .xftp {
+                    return revokeCancelTexts
+                } else {
+                    return nil
+                }
+            case .sndCancelled: return nil
+            case .sndError: return nil
+            case .rcvInvitation: return nil
+            case .rcvAccepted: return rcvCancelTexts
+            case .rcvTransfer: return rcvCancelTexts
+            case .rcvCancelled: return nil
+            case .rcvComplete: return nil
+            case .rcvError: return nil
             }
         }
     }
 }
 
+private var sndCancelTexts = (
+  NSLocalizedString("Stop file", comment: "cancel file action"),
+  NSLocalizedString("Stop sending file?", comment: "cancel file question"),
+  NSLocalizedString("Sending file will be stopped.", comment: "cancel file text"),
+  NSLocalizedString("Stop", comment: "cancel file confirmation")
```

Comment:
these can be struct, and also LocalizedStringKey - there is no need to use NSLocalizedString here I think? At least for the last three, maybe the first needs to be NSLocalizedString, not sure.